tests/ui/len_without_is_empty.rs

```diff
@@ -282,4 +282,17 @@ impl AsyncLen {
     }
 }
 
+// issue #9520
+pub struct NonStandardLenAndIsEmptySignature;
```

Comment:
Would it be possible to add the reproducer from #9520 as a test case?

Reply:
Yep, let me add that in.

Reply:
added